BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/Model/Submission.java

```diff
@@ -0,0 +1,172 @@
+
+    private int face; // the index of the image chosen by user
+    private int thorax; // the index of the image chosen by user
+    private int abdomen; // the index of the image chosen by user
+    private Species species = null;
+    private Habitat habitat = null;
+    private Weather weather = null;
+    private Location location = null;
+    // TODO: Store image in camera role. https://github.com/uwblueprint/foe/issues/22
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof Submission)) {
+            return false;
+        }
+
+        Submission that = (Submission) other;
+
+        return this.face == that.face
+                && this.thorax == that.thorax
+                && this.abdomen == that.abdomen
+                && this.species == that.species
+                && this.habitat == that.habitat
+                && this.weather == that.weather
+                && ((this.location == null && that.location == null) || this.location.equals(that.location));
+    }
+
+    public enum Species {
+        Apidae, Andrenidae, Halictidae, Megachilidae, Colletidae, Unknown
+    }
+
+
+    public enum Habitat {
+        HouseGarden, Park, Swamp, PublicGarden, Lake, Lawn
+    }
+
+    public enum Weather {
+        Sunny, PartlyCloudy, Cloudy, Raining, Hailing
+    }
+
+    public void clear() {
+        face = -1;
+        thorax = -1;
+        abdomen = -1;
+        species = null;
+        habitat = null;
+        weather = null;
+        location = null;
+    }
+
+    public Submission() {
+        clear();
+    }
+
+    public void setFace(int f) throws MissingFormatArgumentException {
+        if (f > MAX_FACE) {
+            throw new MissingFormatArgumentException("Invalid face value");
+        }
+        face = f;
+    }
+
+    public int getFace() {
+        return face;
+    }
+
+    public void setThorax(int t) throws MissingFormatArgumentException {
+        if (t > MAX_THORAX) {
+            throw new MissingFormatArgumentException("Invalid thorax value");
+        }
+        thorax = t;
+    }
+
+    public int getThorax() {
+        return thorax;
+    }
+
+    public void setAbdomen(int a) throws MissingFormatArgumentException {
+        if (a > MAX_ABDOMEN) {
+            throw new MissingFormatArgumentException("Invalid abdomen value");
+        }
+        abdomen = a;
+    }
+
+    public int getAbdomen() {
+        return abdomen;
+    }
+
+    public void setHabitat(Habitat h) {
+        habitat = h;
+    }
+
+    public Habitat getHabitat() {
+        return habitat;
+    }
+
+    public void setWeather(Weather w) {
+        weather = w;
+    }
+
+    public Weather getWeather() {
+        return weather;
+    }
+
+    public void setLocation(Location location) {
+        this.location = location;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+
+    public Species getSpecies() throws MissingFormatArgumentException {
```

Comment:
I don't think there is a real need to separate findSpecies and getSpecies. The logic below isn't worth caching over the readability confusion.

This should probably be the signature
`public Species getSpecies()`